mantidimaging/gui/widgets/spectrum_widgets/fitting_display_widget.py

```diff
@@ -13,14 +14,53 @@ class FittingDisplayWidget(QWidget):
     def __init__(self, parent: QWidget | None = None) -> None:
         super().__init__(parent)
 
-        self.layout: QVBoxLayout = QVBoxLayout(self)
-        self.spectrum_plot: SpectrumPlotWidget = SpectrumPlotWidget()
+        self.layout = QVBoxLayout(self)
+        self.spectrum_plot = SpectrumPlotWidget()
         self.layout.addWidget(self.spectrum_plot)
 
+        self.fitting_region = RectROI([0, 0], [1, 1], pen=mkPen((255, 0, 0), width=2), movable=True)
+        self.fitting_region.setZValue(10)
+        self.fitting_region.addScaleHandle([1, 1], [0, 0])
+        self.fitting_region.addScaleHandle([0, 0], [1, 1])
+        self.fitting_region.addScaleHandle([0, 1], [1, 0])
+        self.fitting_region.addScaleHandle([1, 0], [0, 1])
+        self.spectrum_plot.spectrum.addItem(self.fitting_region)
+
     def update_plot(self, x_data: np.ndarray, y_data: np.ndarray, label: str = "ROI") -> None:
+        if x_data is None or x_data.size == 0:
+            return
         self.spectrum_plot.spectrum.clear()
         self.spectrum_plot.spectrum.plot(x_data, y_data, name=label, pen=(255, 255, 0))
+        self.spectrum_plot.spectrum.addItem(self.fitting_region)
```

Comment:
`self.fitting_region` probably does need to be added each time the plot is updated. Maybe just once in the constructor.